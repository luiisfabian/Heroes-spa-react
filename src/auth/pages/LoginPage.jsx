import React from 'react'
import { useContext } from 'react';
import { useNavigate } from 'react-router-dom'
import { authContext } from '../context/AuthContext';

export const LoginPage = () => {


  const navigate = useNavigate();
  const { login } = useContext(authContext)



  const onLogin = () => {

    const lastPash = localStorage.getItem('lastPath') || '/'
    login('Fabian Mu√±oz');

    navigate(lastPash, {
      replace: true
    })
  }

  return (
    <>

      <div className="container mt-5">
        <h1>LoginPage</h1>
        <hr />
        <button className='btn btn-primary' type="button" onClick={onLogin}>Ingresar</button>
      </div>
    </>
  )
}
