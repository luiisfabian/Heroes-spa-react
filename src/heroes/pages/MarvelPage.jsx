import { HeroList } from "../components"

export const MarvelPage = () => {
  return (
    <>
    <h1>MARVEL Comics</h1>
    <hr />
    <HeroList publisher="Marvel Comics" />
    </>
  )
}
