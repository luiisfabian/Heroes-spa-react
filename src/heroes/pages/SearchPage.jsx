import { HeroCard } from "../components/HeroCard"

export const SearchPage = () => {
  return (
    <>
      <h1>SearchPage</h1>
      <hr />
      <div className="row">
        <div className="col-5">
          <h4>Searching</h4>
          <hr />
          <form action="">
            <input type="text" placeholder="Search Hero" className="form-control" name="SearchText" autoComplete="off" />
            <button className="btn btn-success mt-1">Search</button>
          </form>
        </div>
        <div className="col-7">
          <h4>Result</h4>
          <hr />

          <div className="alert alert-primary">
            Search a Hero!
          </div>

          <div className="alert alert-danger">
            <b>No hero Result</b>
          </div>

          {/* <HeroCard  /> */}
        </div>
      </div>


    </>
  )
}
