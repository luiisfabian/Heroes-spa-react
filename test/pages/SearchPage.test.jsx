import { fireEvent, render, screen } from "@testing-library/react";
import { SearchPage } from "../../src/heroes/pages/SearchPage";
import { MemoryRouter } from "react-router-dom";


const mockedUseNavigate = jest.fn()

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUseNavigate,
}))

describe('test in SearchPage', () => {
    test('debe mostrarse correctamente con los valores por defecto', () => {
        // beforeEach(() => jest.clearAllMocks())

        const { container } = render(
            <MemoryRouter>,
                <SearchPage />
            </MemoryRouter>
        )
        // expect(container).toMatchSnapshot()
        expect(screen.getByText('Search')).toBeTruthy()
        expect(screen.getByText('Search a Hero!')).toBeTruthy()
    })

    test('debe mostrar batman y con el input del valor del query string', () => {
        render(
            <MemoryRouter initialEntries={['/search?q=batman']}>,
                <SearchPage />
            </MemoryRouter>
        )
        const input = screen.getByRole('textbox');
        expect(input.value).toBe("batman")

        const img = screen.getByRole('img');

        expect(img.src).toContain("http://localhost/assets/heroes/dc-batman.jpg")

        const alert = screen.getByLabelText('alert-danger');
        expect(alert.style.display).toBe('none')
    })

    test('debe de mostrar un error si no se encuentra el herue', () => {
        render(
            <MemoryRouter initialEntries={['/search?q=batman123']}>,
                <SearchPage />
            </MemoryRouter>
        )
        const alert = screen.getByLabelText('alert-danger');
        expect(alert.style.display).toBe('')
        // expect(input.value).toBe("No hero Result")
    })
    test('debe de llamar al navigate a una pestaÃ±a nueva', () => {
        const inputValue = 'superman'
        render(
            <MemoryRouter initialEntries={['/search']}>,
                <SearchPage />
            </MemoryRouter>
        )
        // screen.debug()
        const input = screen.getByRole('textbox')

        fireEvent.change(input, { name: 'searchText', value: inputValue })
        const form = screen.getByRole('form')
        fireEvent.submit(form)
        expect( mockedUseNavigate ).toHaveBeenCalledWith(`?q=${ inputValue }`)
    
    
    })
});