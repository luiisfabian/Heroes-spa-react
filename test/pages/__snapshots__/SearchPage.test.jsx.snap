// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in SearchPage debe mostrar batman y con el input del valor del query string 1`] = `
<div>
  ,
  <h1>
    SearchPage
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search Hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-success mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Result
      </h4>
      <hr />
      <div
        class="alert alert-primary"
      >
        Search a Hero!
      </div>
      <div
        class="alert alert-danger"
        style="display: none;"
      >
        <p>
          No hero Result 
          <b>
             
          </b>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`test in SearchPage debe mostrarse correctamente con los valores por defecto 1`] = `
<div>
  ,
  <h1>
    SearchPage
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search Hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-success mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Result
      </h4>
      <hr />
      <div
        class="alert alert-primary"
      >
        Search a Hero!
      </div>
      <div
        class="alert alert-danger"
        style="display: none;"
      >
        <p>
          No hero Result 
          <b>
             
          </b>
        </p>
      </div>
    </div>
  </div>
</div>
`;
